src/components/virtual-scroll/test/basic/app.module.ts

```diff
@@ -54,8 +42,20 @@ export class E2EPage {
     this.counter++;
   }
 
-  reload() {
-    window.location.reload(true);
+  addRandomItem() {
+    const index = Math.floor(Math.random() * this.items.length);
+    this.items.splice( index, 0, {
+      value: Math.floor(Math.random() * 10000),
+      someMethod: function() {
+        return '!!';
+      }
+    }
+    );
+  }
+
+  changeItem() {
+    const index = Math.floor(Math.random() * this.items.length);
+    this.items[index].value = Math.floor(Math.random() * 10000);
```

Comment:
This does not change the item reference and thus is not expected to generate a diff.
```
    this.items[index] = { value: Math.floor(Math.random() * 10000), someMethod: () => '!!' };
```
is more appropriate.